htsget-config/src/config/mod.rs

```diff
@@ -0,0 +1,755 @@
+pub mod cors;
+
+use std::fmt::Debug;
+use std::io;
+use std::io::ErrorKind;
+use std::net::SocketAddr;
+use std::path::{Path, PathBuf};
+
+use crate::config::cors::{AllowType, CorsConfig, HeaderValue, TaggedAllowTypes};
+use clap::Parser;
+use figment::providers::{Env, Format, Serialized, Toml};
+use figment::Figment;
+use http::header::HeaderName;
+use http::Method;
+use serde::{Deserialize, Serialize};
+use serde_with::with_prefix;
+use tracing::info;
+use tracing::instrument;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::{fmt, EnvFilter, Registry};
+
+use crate::regex_resolver::RegexResolver;
+
+/// Represents a usage string for htsget-rs.
+pub const USAGE: &str =
+  "htsget-rs can be configured using a config file or environment variables. \
+See the documentation of the htsget-config crate for more information.";
+
+const ENVIRONMENT_VARIABLE_PREFIX: &str = "HTSGET_";
+
+pub(crate) fn default_localstorage_addr() -> &'static str {
+  "127.0.0.1:8081"
+}
+
+fn default_addr() -> &'static str {
+  "127.0.0.1:8080"
+}
+
+fn default_server_origin() -> &'static str {
+  "http://localhost:8080"
+}
+
+pub(crate) fn default_path() -> &'static str {
+  "data"
```

Comment:
Is this the standard default path on the ref impl or there's no equivalent?

Reply:
I don't think there is an equivalent. This is default so that `cargo run -p htsget-http-actix` correctly points to the `data` folder in our repo with all the test files. I think the ref impl just finds the local files by resolving the equivalent of our regex resolver and substitution string. Perhaps this is the better approach as it is a little bit less complicated?